compiler/rustc_lexer/src/unescape.rs

```diff
@@ -55,6 +55,9 @@ pub enum EscapeError {
     /// Non-ascii character in byte literal, byte string literal, or raw byte string literal.
     NonAsciiCharInByte,
 
+    /// A NUL character was found in C string literal.
+    NulInCStr,
```

Comment:
This should be a semantic error, really.
At token level we shouldn't care whether the string contains a zero or not, it's a property that is only important at runtime.
In other words
```rust
#[cfg(FALSE)]
let s = c"\0";
```
should work.

Reply:
I think I see the point now. Will do that when I get to it.

Reply:
Yeah see also #102944